[FrameworkBundle] Fix sorting bug in sorting of tagged services by priority

Ahummeling opened the pull request:
Fixed incorrect assumption that the minimum priority is zero
leading to an unsorted list of negative priorities.

| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no 
| Tickets       | Fix #45396
| License       | MIT
| Doc PR        | 

Fixed an incorrect assumption that the minimum priority is zero which was leading to an unsorted list of negative priorities.
(See #45396)

Ahummeling committed:
fix sorting bug

Fixed incorrect assumption that the minimum priority is zero
leading to an unsorted list of negative priorities.

dkarlovi commented:
@Ahummeling if I understood correctly, it means the positive priorities are sorted correct, but negative are not? 

Is `sortTaggedServicesByPriority` used to actually inject the services into the serializer? It seems unfortunate we'd have one method sorting them for display and another sorting them for injection, they'd always be in danger to go out of sync. :thinking:

Ahummeling commented:
> @Ahummeling if I understood correctly, it means the positive priorities are sorted correct, but negative are not?
> 
> Is `sortTaggedServicesByPriority` used to actually inject the services into the serializer? It seems unfortunate we'd have one method sorting them for display and another sorting them for injection, they'd always be in danger to go out of sync. thinking

Yes you are understanding this correctly.

The class responsible for sorting the output of the debug command is the `Symfony\Bundle\FrameworkBundle\Console\Descriptor\Descriptor`. I can't imagine this is used in the serializer, which would explain the differences.

Ahummeling commented:
The `SerializerPass` seems to call `Symfony\Component\DependencyInjection\Compiler\PriorityTaggedServiceTrait::findAndSortTaggedServices`
which has a completely different algorithm for sorting the services, but I believe it still initializes default priority to 0 in the case of the serializers. I am not quite sure how it works exactly, as it's quite a bit more complicated

nicolas-grekas commented:
Can you please add a test case?

Ahummeling commented:
> Can you please add a test case?

Certainly, sorry should've done that before making the PR

Ahummeling commented:
Added a test case to the relevant provider that would fail without the proposed change and updated the relevant fixtures to reflect this additional testcase.

nicolas-grekas commented:
Thank you @Ahummeling.
